src/peft/tuners/tuners_utils.py

```diff
@@ -0,0 +1,203 @@
+# coding=utf-8
+# Copyright 2023-present the HuggingFace Inc. team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from abc import ABC, abstractmethod
+
+from torch import nn
+
+from ..config import PeftConfig
+from ..utils import _get_submodules
+
+
+class BaseTuner(nn.Module, ABC):
+    r"""
+    A base tuner model that provides the common methods and attributes for all tuners. Each `BaseTuner` class needs to
+    be implemented in case the adapter is a plugable adapter.
+    """
+
+    def __init__(self, model, peft_config, adapter_name):
+        super().__init__()
+
+        self.model = model
+        self.forward = self.model.forward
+
+        # Some adapters might be already attached
+        if not hasattr(self, "peft_config"):
+            self.peft_config = peft_config
```

Comment:
I am uncertain about when the `else` case could happen: If there is already a `peft_config` attribute, the `peft_config` being passed here is just ignored. Is this something that could be confusing to users? Or maybe we should check that the existing `peft_config` is identical to the one being passed here and issue a warning if not?

Reply:
Hmm actually this condition is not needed for PEFT but I added it as a hack to make it work for the transformers integration + multiple adapters. 

`model.load_adapter` will re-create new PeftModel instance and change the current model in-place. If I don't add that hack it will overwrite the current existing peft_config dict with a new one

Perhaps I can elaborate this in the comments or add a warning telling users that there is already a `peft_config` attached to the model ?

Reply:
I see. So if this is the only situation where this occurs, I think a comment is enough.

Reply:
Added an info and more comments